flang/lib/Lower/OpenMP/PrivateReductionUtils.cpp

```diff
@@ -472,8 +493,8 @@ void PopulateInitAndCleanupRegionsHelper::initAndCleanupBoxedArray(
   // Put the temporary inside of a box:
   // hlfir::genVariableBox doesn't handle non-default lower bounds
   mlir::Value box;
-  fir::ShapeShiftOp shapeShift =
-      getShapeShift(builder, loc, getLoadedMoldArg());
+  fir::ShapeShiftOp shapeShift = getShapeShift(builder, loc, getLoadedMoldArg(),
```

Comment:
Does loading `moldArg` count as an use?
Or will we only insert a barrier if the loaded `moldArg` is referenced?

Reply:
Judging by the code in https://github.com/llvm/llvm-project/pull/125900, if there is only a load of `moldArg` it will be removed and no barrier will be inserted.